semantics/language/common/expr/eval.k

```diff
@@ -41,19 +43,19 @@ module C-COMMON-EXPR-EVAL
      }*/
 
      // Lvalue conversion.
-     rule reval(nclv(Loc:SymLoc, T:Type)
+     rule reval(nclv(Loc::SymLoc, T::Type)
```

Comment:
So Grigore reminded me that we have strict compilation on by default. I did some preliminary testing and it looks like if you don't use the --non-strict option, sort checks should get automatically inferred in cases like this, even when the user types `::` instead of `:`. Thus, you should be able to simply disable that flag and get the original behavior, which seems like a sufficient compromise if you are concerned about backends where increased sort information is desirable.